mlflow/xgboost.py

```diff
@@ -416,10 +443,51 @@ def log_feature_importance_plot(features, importance, importance_type):
             finally:
                 shutil.rmtree(tmpdir)
 
-        try_mlflow_log(log_model, model, artifact_path="model")
+        # dtrain must exist as the original train function already ran successfully
+        dtrain = args[1] if len(args) > 1 else kwargs.get("dtrain")
+
+        input_example_info = None
+        input_example = None
+        signature = None
+        try:
+            # it is possible that the dataset was constructed before the patched
+            #   constructor was applied, so we cannot assume the input_example_info exists
+            input_example_info = getattr(dtrain, "input_example_info", None)
+
+            if input_example_info is None:
+                raise Exception(
+                    "please ensure that autologging is "
+                    + "enabled before constructing the dataset."
+                )
+
+            input_example = input_example_info.input_example
+            if input_example is None:
+                # input example collection failed
+                raise Exception(input_example_info.error_msg)
+        except Exception as e:  # pylint: disable=broad-except
+            _logger.warning("Failed to gather example input: " + str(e))
+
+        if input_example is not None:
+            # input example collection succeeded, move on to signature prediction
+            try:
+                model_output = model.predict(xgb.DMatrix(input_example))
+                signature = infer_signature(input_example, model_output)
+            except Exception as e:  # pylint: disable=broad-except
+                input_example = None
+                msg = "Failed to infer the model signature: " + str(e)
+                _logger.warning(msg)
```

Comment:
Can we consolidate this into the same try/catch block that's used for input example extraction? Ultimately, the user either ends up with both input example and signature or ends up with neither one. We could use a single warning message beginning with `Failed to gather example input and model signature:` to make this clear. Otherwise, if the input example is missing, we don't currently say anything about the signature being absent in our warning message.